json-rpc/src/runtime.rs

```diff
@@ -291,6 +363,14 @@ fn set_response_error(response: &mut JsonRpcResponse, error: JsonRpcError) {
     response.error = Some(error);
 }
 
+fn is_internal_error(error: Option<JsonRpcError>) -> bool {
+    if let Some(e) = error {
+        e.code <= -32000 && e.code >= -32099
+    } else {
+        false
+    }
+}
```

Comment:
You should probably be able to pass in a reference to the `Option<JsonRpcError>` and then you won't need to clone it.

Reply:
Specifically you'll probably want the signature to be `Option<&JsonRpcError>` instead of `&Option<JsonRpcError>`